pkg/process/checks/process_linux.go

```diff
@@ -52,17 +56,153 @@ func (p *ProcessCheck) processesByPID(collectStats bool) (map[int32]*procutil.Pr
 }
 
 func mapWLMProcToProc(wlmProc *workloadmetacomp.Process, stats *procutil.Stats) *procutil.Process {
+	var service *procutil.Service
+	var ports []uint16
+	if wlmProc.Service != nil {
+		service = &procutil.Service{
+			GeneratedName:            wlmProc.Service.GeneratedName,
+			GeneratedNameSource:      wlmProc.Service.GeneratedNameSource,
+			AdditionalGeneratedNames: wlmProc.Service.AdditionalGeneratedNames,
+			TracerMetadata:           wlmProc.Service.TracerMetadata,
+			DDService:                wlmProc.Service.DDService,
+			APMInstrumentation:       wlmProc.Service.APMInstrumentation,
+		}
+		ports = wlmProc.Service.Ports
+	}
 	return &procutil.Process{
-		Pid:     wlmProc.Pid,
-		Ppid:    wlmProc.Ppid,
-		NsPid:   wlmProc.NsPid,
-		Name:    wlmProc.Name,
-		Cwd:     wlmProc.Cwd,
-		Exe:     wlmProc.Exe,
-		Comm:    wlmProc.Comm,
-		Cmdline: wlmProc.Cmdline,
-		Uids:    wlmProc.Uids,
-		Gids:    wlmProc.Gids,
-		Stats:   stats,
+		Pid:      wlmProc.Pid,
+		Ppid:     wlmProc.Ppid,
+		NsPid:    wlmProc.NsPid,
+		Name:     wlmProc.Name,
+		Cwd:      wlmProc.Cwd,
+		Exe:      wlmProc.Exe,
+		Comm:     wlmProc.Comm,
+		Cmdline:  wlmProc.Cmdline,
+		Uids:     wlmProc.Uids,
+		Gids:     wlmProc.Gids,
+		Stats:    stats,
+		Ports:    ports,
+		Language: wlmProc.Language,
+		Service:  service,
+	}
+}
+
+// formatPorts converts a list of uin16 ports to a int32 PortInfo
+// TODO: because the service discovery response does not distinguish between tcp and udp currently, we currently send everything as TCP
```

Comment:
For agent discovery folks: Why are they not distinguished? As far as I understand, depending on how we want to present this information on Datadog, this can cause an issue where old agents will show UDP connections as TCP.

Reply:
It wasn't a requirement to do so for SD but since we plant to be use this for things like autodiscovery it makes sense to be want to know which protocol the ports are on. I can send a follow up or separate PR to change the collection to distinguish them.

Reply:
I think that would be great. Would make things simpler on the backend so we don't ever need to worry about if the ports are distinguished by the agent or not.